blocks/math.ts

```diff
@@ -577,13 +560,11 @@ const LIST_MODES_MUTATOR_MIXIN = {
 /**
  * Extension to 'math_on_list' blocks that allows support of
  * modes operation (outputs a list of numbers).
- * @this {Block}
- * @package
  */
-const LIST_MODES_MUTATOR_EXTENSION = function() {
-  this.getField('OP').setValidator(function(newOp) {
-    this.updateType_(newOp);
-  }.bind(this));
+const LIST_MODES_MUTATOR_EXTENSION = function(this: Block) {
+  this.getField('OP')!.setValidator(function(this: Field, newOp: string) {
```

Comment:
Could the validator be converted to an arrow function to avoid having to explicitly bind this/generally modernize it?

Reply:
I don't think so, because the validator makes use of `this`, and if I understand correctly an arrow expression here would have `this` irrevocably bound to `LIST_MODES_MUTATOR_EXTENSION`, which is not what we want.

(If only block definitions were classes, and these could just be ordinary methods!)

Reply:
(No, now that I think more carefully I think that's wrong: `this` would be `undefined`—still wrong.)

Reply:
The more I look at this bit of code the weirder it is.  I think  I will rewrite it—but in a future PR.  For now just try to preserve existing behaviour exactly.